Assets/_BForBoss/_Character/Scripts/FirstPersonPlayer.cs

```diff
@@ -6,16 +6,15 @@
 
 namespace BForBoss
 {
-    public class FirstPersonPlayer : FirstPersonCharacter
+    public partial class FirstPersonPlayer : FirstPersonCharacter
```

Comment:
What is the reason for using a partial class and not just allowing `FirstPersonPlayer` to inherit from `ICharacterSpeed`?

Reply:
So this class here is our base `FirstPersonPlayer` that does what it needs. 
So if we copy this class to another project, it'll work as intended. 

And then underneath (the other partial), are all the extra methods that it needs to implement so I could inject it into other modules else where i.e. Whatever UI that needs specific data from this class.

So if we were to move this class to FNF, we only need to move this partial class, without the extra `ICharacterSpeed` stuff. 

hope this makes sense.

Reply:
Ideally, the partial class: `ICharacterSpeed` would be in another file. But I was just lazy.